Controller/Adminhtml/Order/ReceivedUrl.php

```diff
@@ -17,47 +17,22 @@
 
 namespace Bolt\Boltpay\Controller\Adminhtml\Order;
 
-use Magento\Framework\App\Action\Action;
 use Magento\Framework\App\Action\Context;
 use Bolt\Boltpay\Helper\Log as LogHelper;
 use Bolt\Boltpay\Helper\Cart as CartHelper;
 use Bolt\Boltpay\Helper\Config as ConfigHelper;
 use Bolt\Boltpay\Helper\Bugsnag;
-use Magento\Framework\Exception\LocalizedException;
-use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Checkout\Model\Session as CheckoutSession;
-use Magento\Quote\Model\Quote;
 use Magento\Sales\Model\Order;
+use Bolt\Boltpay\Helper\Order as OrderHelper;
 
 /**
  * Class ReceivedUrl
  *
  * @package Bolt\Boltpay\Controller\Adminhtml\Order
  */
-class ReceivedUrl extends Action
+class ReceivedUrl extends \Bolt\Boltpay\Controller\Order\ReceivedUrl
```

Comment:
Usually for adminhtml controllers uses the class `Magento\Backend\App\Action`.
`\Bolt\Boltpay\Controller\Order\ReceivedUrl` - did not extend from adminhtml so we do not have correct session, permissions and etc.

Reply:
@YevhenShBolt 
all our admin controllers use Magento\Framework\App\Action instead of Magento\Backend\App\Action. This works and we never had the problems with session or permissions. This was probably the workaround for maintaining the form key for call made from the checkout javascript. Anyway I tested this and it works. Using Magento\Backend\App\Action had thrown errors last week. I'd leave this like this for now and see what can be done in the future.

Reply:
@danac-gs.
Some code or vulnerability can call our class from frontend and all dependencies will load and we lost security here. We do not have any session check or auth check from admin side now (but in backend action it exists).
We should refactor it now (or create a ticket for it and make refactoring after...)because in some time we can have a security problem.

Reply:
@YevhenShBolt what you said is correct and valid.
However, in practice, with the particular calls we make I don't see the problem. Let me explain. 

We only have 2 admin controllers, Cart\Data and Order\ReceivedUrl. Although Order\Save is still there, it is redundant with the pre-auth flow.

Cart\Data creates the cart / immutable quote, operating exclusively based on the session related data, can not be called out of the admin session (it can but it can not find the quote), has no potentially malicious side effect on the parent quote, the only input that it can take is the billing address sent to bolt (this might be redundant, it is there for payment only from front-end that no one uses and our support for that is not confirmed in the past year).

The Order\ReceivedUrl input is generated and hashed by bolt and verified via signing secret. 

Nevertheless, I will create a task to investigate and refactor this.

Reply:
@danac-gs 
ok. it is good for me with ticket.
Below left  one more review.

Reply:
I also made Bolt\Boltpay\Controller\Order\ReceivedUrl methods private